CONTRIBUTING.md

```diff
@@ -423,6 +423,11 @@ the following steps:
 The `stable/*` branches should only receive changes in the form of bug
 fixes.
 
+## Adding deprecation warnings
+The qiskit-terra code is part of Qiskit and, therefore, the [Qiskit Deprecation Policy](https://qiskit.org/documentation/contributing_to_qiskit.html#deprecation-policy) fully applies here. Additionally, qiskit-terra does not allow `DeprecationWarning`s in its testsuite. If you are deprecating code, considering adapting the test to use the new/non-deprecated method or, in case you want to still check that the deprecated method works as expected, [use `assertWarns`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns). The `assertWarns` context will silence the deprecation warning while checking that it raises.
```

Comment:
Shouldn't we be recommending for people to write additional tests for new/non-deprecated methods instead of overwriting the existing tests? I would have thought the appropriate process would be to write new tests for new methods AND refactor deprecated method tests with assertWarns. That way both new and deprecated code is covered, until the deprecated code (and it's corresponding test) is removed

Reply:
true. I changed the wording and added an example.